### PR TITLE
Form test harness

### PR DESCRIPTION
**Pull request by markacianfrani**
## Brief Description

Sets up native form integration E2E tests in Cypress to be completed in future tickets.

You can test by renaming `form-select.e2e.example` to `form-select.e2e.js` and running `npm run cy.test.watch`. Select currently passes because its not shadow dom encapsulated. All other components currently fail.

Example tests include our astro component and its native component counterpart. The test interacts with each element the same way and submits the form. The form data request is intercepted and displayed in the body as a quick way to see and test what was submitted. Easiest way I could think of to test FormData, but this will probably need to be iterated on for edge cases. 

Includes the renderHiddenInput utility from Ionic

**Tests are not 100% perfect. Will require dev work when it comes to actually working on them. They are more for general guidance. **

## JIRA Link

[ASTRO-1654](https://rocketcom.atlassian.net/browse/ASTRO-1654)

## Motivation and Context

Our form web components need to work with native `<form>` elements in plain HTML, no build process, no javascript. 

## Issues and Limitations

* The github CI/CD for running Cypress will need to be fixed in a separate ticket. Tests are named .example to avoid having them fail in the PR check temporarily.
* Cypress was currently setup to run from Storybook, which works great for single component tests, but we need to test multiple components here. It doesn't make sense to clutter our Storybook with automated tests so I needed to modify the test runner to run off of Stencil's dev server. 
* Does the directory structure make sense? We just need a place for custom non-isolated E2E tests. I couldn't think of a great name.
* Is there a better way to test FormData? 

## Types of changes

-   [ ] Bug fix
-   [x] New feature
-   [ ] Breaking change

## Checklist

-   [x] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

File: src/tests/pages/form-checkbox.html [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    id="nativeCheckbox"
+                    type="checkbox"
+                    value="foo"
+                    name="nativeCheckbox"
+                />
+                <button type="submit">submit</button>
+            </form>
+            <ul id="log"></ul>
+            <script>
+                const log = document.getElementById('log')
```
Comment (markacianfrani) [inline src/tests/pages/form-checkbox.html:44]: would be nice if we could abstract this into some helper util, but I can't think of a great way without adding unnecessary code to our build folder.

File: stencil.config.ts [hunk 2]
```diff
@@ -26,7 +26,12 @@ export const config: Config = {
         },
         {
             type: 'www',
-            serviceWorker: null, // disable service workers
+            serviceWorker: null, // disable service workers,
+            copy: [
+                {
+                    src: 'tests/pages',
```
Comment (markacianfrani) [inline stencil.config.ts:32]: responsible for creating the isolated index.html pages at localhost/tests/pages/foo.html

**Review by Cheshire89 — APPROVED** [review]

**Review by micahjones13 — APPROVED** [review]
